Wrote a results paragraph, minor comments and fixes (#3)
big-data-lab-team/MCA-libmath-paper: paper.tex

reproducibility of numerically unstable computational pipelines. In
particular, operating system (OS) updates frequently involve mathematical
libraries, resulting in the computation of slightly different albeit

> equivalently correct results across OS versions. In practice, this has important

Reviewer: Comment shifted to the paper.

given subject was only about 4 times lower than the average RMSE between
cluster representatives of different subjects,
which indicates that numerical and anatomical variabilities were in the
same order of magnitude. Notably, the similarity between subjects was
also preserved by the numerical perturbation, leading to the same subject
ordering in the dendrograms in Figure~\ref{fig:clusters} except for one inversion.

\begin{figure}
  \centering
  \includegraphics[width=\columnwidth]{figures/Hclustering.png}
  \caption{RMSE-based hierarchical clustering of OS (left) and MCA
    (right) samples. Colors identify different subjects, showing that
    similarities between subjects are preserved by the numerical perturbations. Horizontal gray lines 
    represent average RMSEs between (top line) and within (bottom line) subject clusters. \TG{can we use a color map where the 20 subjects 
    would be distinguishable? For suggestions see \url{https://personal.sron.nl/~pault/\#fig:scheme\_rainbow\_discrete}}}
  \label{fig:clusters}
\end{figure}

\subsection{Results are stable across virtual precisions}

The results presented previously were obtained at the virtual precision of

> t=25~bits, determined as the one minimizing the RMSE between MCA and OS

Reviewer 2: One detail that @yohanchatelain and I discussed: should we really be going with 25, then the plateau is so flat for so long? In the selection of "nearest" it might make sense to have a penalty term that becomes smaller as you get higher precision, which would push us towards 35ish. Given that the 10 closest values are all nearly-equivalent, it just feels like we need to justify our strict obedience of the SSD in this noisy signal, while if we added a penalty term we would have a much more justifiable value.

maps of significant digits. We varied the virtual precision by steps of 2
between t=1~bit and t=53~bits (Fig.~\ref{fig:vprecision}). On average, no
noticeable RMSE change was observed between the MCA and OS variability for
precisions ranging from t=3~bits to t=35~bits. This observation suggests
that the accurate simulation of OS-related variability by MCA is stable

> across a wide range of virtual precisions, which is practical. The RMSE of

Reviewer 2: ```suggestion
across a wide range of virtual precisions. The RMSE of
```

individual subjects in this virtual precision range is consistent with the
average line, with the notable exception of subjects 5 and 15 \TG{Ali,
check subject numbers} \TG{Ali, could you show the sig digits difference

> map (MCA vs OS) for these subjects at t=29 and t=31?}. The jitsy behavior

Reviewer 2: ```suggestion
map (MCA vs OS) for these subjects at t=29 and t=31?}. The jittery behaviour
```

of subject 15 suggests the existence of multiple local minima in the atlas
registration \TG{check that}, possibly reached by different OS versions and/or MCA samples.
Beyond the virtual

> precision of t=35~bits, subjects are split in two groups: for subjects

Reviewer 2: ```suggestion
precision of t=35~bits, subjects appear to split in two groups: subjects
```

\TG{x, y and z}, MCA samples remain accurate proxies for OS variability;

> for subjects \TG{a, b, and c}, it was not the case, suggesting that the

Reviewer 2: ```suggestion
however, for subjects \TG{a, b, and c} this was not the case, suggesting that the
```

> perturbation introduced by MCA was not important enough to drag the result

Reviewer 2: ```suggestion
perturbation introduced by MCA was not significant enough to drag the result
```

> to the local minima reached in other OS versions. The observed discrepancies between subjects 

Reviewer 2: ```suggestion
to the local minima reached in another OS version. The observed discrepancies between subjects 
```

> might be leveraged in quality control checks and as a result to inform tool development.

Reviewer 2: ```suggestion
might be leveraged in quality control checks and, as a result, inform tool development.
```
